Analytics/Request/BlockingRequestHandler.cs

```diff
@@ -225,19 +225,28 @@ public async Task MakeRequest(Batch batch)
                     if (_client.Config.CompressRequest)
                         content.Headers.ContentEncoding.Add("gzip");
 
-                    var response = await _httpClient.PostAsync(uri, content).ConfigureAwait(false);
+                    HttpResponseMessage response = null;
+                    bool retry = false;
+                    try
+                    {
+                        response = await _httpClient.PostAsync(uri, content).ConfigureAwait(false);
+                    }
+                    catch (System.Threading.Tasks.TaskCanceledException)
```

Comment:
Is there a way to catch HttpRequestException here (https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.postasync?view=netframework-4.7.2)?

Reply:
There's no way to catch HttpReqeustException in this case.

Reply:
If HttpRequestException occurs in PostAsync, await operation will be canceled.
So I used TaskCanceledException to catch error.

Reply:
The concern with catching TaskCanceledException is that it will catch other errors as well, but we are really only interested in network errors (like HttpRequestException). Maybe we can check the inner exception? https://docs.microsoft.com/en-us/dotnet/api/system.exception.innerexception?view=netframework-4.7.2

Reply:
It doesn't contain InnerException.